### PR TITLE
Use intmin for invalid pixel positions, allow empty arguments

### LINKED ISSUE #1828: CameraGeometry position_to_pix_index returns -1 for nan input
https://github.com/cta-observatory/ctapipe/blob/39762693534f7dc9b884684c3f82e153bd9da560/ctapipe/instrument/camera/geometry.py#L859

For example, of one uses it on image centroid coordinates to make a "pixelized" 2d histogram, all non-reconstructed images end up in the highest-index pixel (=-1). It would be better to return np.nan.

### PR DESCRIPTION
**Pull request by maxnoe**
Fixes #1828

### CODE REVIEW

File: docs/changes/2397.api.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+``CameraGeometry.position_to_pix_index`` will no return the minimum integer value for invalid
```
Comment (Tobychev) [inline docs/changes/2397.api.rst:1]: should it be "now" instead of no?

File: docs/changes/2397.api.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+pixel coordinates instead of -1 due to the danger of using -1 as an index in python accessing
+the last element of a data array for invalid pixels.
+The function will no also no longer raise an error if the arguments are empty arrays and instead
```
Comment (Tobychev) [inline docs/changes/2397.api.rst:4]: should the first no be "now" instead?

File: docs/changes/2397.api.rst [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+also just return an empty index array.
```
Comment (Tobychev) [inline docs/changes/2397.api.rst:5]: not sure if the "also" should be here, the function always returns one variable, not the the optional extra value implied by "also".

**Comment by Tobychev** [comment]
Also, this seems to have encountered that same problem with traitlets and the docs that you fixed for #2396

**Comment by maxnoe** [comment]
@Tobychev yes, thanks, I opened a PR fixing that so we can get tht in fast: #2399

**Review by Tobychev — APPROVED** [review]

**Review by LukasNickel — APPROVED** [review]

**Pull request merged**